source/material_model/viscoelastic_plastic.cc

```diff
@@ -0,0 +1,737 @@
+                             "For problems with a variable viscosity, the reference viscosity "
+                             "should be a value that adequately represents the order of "
+                             "magnitude of the viscosities that appear, such as an average "
+                             "value or the value one would use to compute a Rayleigh number."
+                             "\n\n"
+                             "Units: $Pa \\, s$");
+
+          // Equation of state parameters
+          prm.declare_entry ("Densities", "3300.",
+                             Patterns::List(Patterns::Double(0)),
+                             "List of densities for background material and compositional fields, "
+                             "for a total of N+1 values, where N is the number of compositional fields. "
+                             "If only one value is given, then all use the same value.  Units: $kg / m^3$");
+          prm.declare_entry ("Thermal expansivities", "4.e-5",
+                             Patterns::List(Patterns::Double(0)),
+                             "List of thermal expansivities for background material and compositional fields, "
+                             "for a total of N+1 values, where N is the number of compositional fields. "
+                             "If only one value is given, then all use the same value. Units: $1/K$");
+          prm.declare_entry ("Specific heats", "1250.",
+                             Patterns::List(Patterns::Double(0)),
+                             "List of specific heats $C_p$ for background material and compositional fields, "
+                             "for a total of N+1 values, where N is the number of compositional fields. "
+                             "If only one value is given, then all use the same value. Units: $J /kg /K$");
+          prm.declare_entry ("Thermal conductivities", "4.7",
+                             Patterns::List(Patterns::Double(0)),
+                             "List of thermal conductivities for background material and compositional fields, "
+                             "for a total of N+1 values, where N is the number of compositional fields. "
+                             "If only one value is given, then all use the same value. Units: $W/m/K$ ");
+          prm.declare_entry ("Viscosity averaging scheme", "harmonic",
+                             Patterns::Selection("arithmetic|harmonic|geometric|maximum composition "),
+                             "When more than one compositional field is present at a point "
+                             "with different viscosities, we need to come up with an average "
+                             "viscosity at that point.  Select a weighted harmonic, arithmetic, "
+                             "geometric, or maximum composition.");
+
+          prm.declare_entry ("Linear viscosities", "1.e21",
```

Comment:
To keep the style of the rest of the parameters please remove the empty line above the parameter.